### PR TITLE
Update #465 with `LocalFNO` and tests

### PR DESCRIPTION
**Pull request by dhpitt**
Supersedes #465. Also removes `implementation` from `FiniteDifferenceConvolution` and `LocalFNOBlocks` due to bugs in 1D and 3D

### CODE REVIEW

**Review by JeanKossaifi — COMMENTED** [review]
Looks good, this builds nicely on the previous PR - can we check the doc gets rendered properly? I'll review the code more in detail asap.

File: neuralop/layers/local_fno_block.py [hunk 1]
```diff
@@ -180,13 +216,21 @@ def __init__(
...
+        self.disco_layers = disco_layers
+        self.disco_kernel_shape = disco_kernel_shape
+        self.radius_cutoff = radius_cutoff
+        self.domain_length = domain_length
+        self.disco_groups = disco_groups
+        self.disco_bias = disco_bias
+        self.periodic = (self.conv_padding_mode in ['circular', 'periodic'])
+
         assert len(diff_layers) == n_layers, "Length of diff_layers must be n_layers"
+        assert len(disco_layers) == n_layers, "Length of disco_layers must be n_layers"
```
[the remark this answers is not in the document]
Comment (dhpitt) [reply neuralop/layers/local_fno_block.py:233]: Sure

File: neuralop/layers/local_fno_block.py [hunk 2]
```diff
@@ -389,7 +466,7 @@ def forward_with_preactivation(self, x, index=0, output_shape=None):
 
         x_fno = self.convs[index](x, output_shape=output_shape)
 
-        x = x_fno + x_skip_fno_diff
+        x = x_fno + x_skip_fno_diff_disco
```
Comment (JeanKossaifi) [inline neuralop/layers/local_fno_block.py:469]: We could potentially just call this `x_skip_local_fno`

**Review by JeanKossaifi — COMMENTED** [review]
It looks good to me, thanks for the changes!

File: neuralop/layers/local_fno_block.py (hunk 2 above)
Comment (dhpitt) [reply neuralop/layers/local_fno_block.py]: That sounds good. Changing

**Comment by JeanKossaifi** [comment]
Great, thanks @dhpitt, merging.

**Pull request merged**